pkg/operator/controller/dns/controller_test.go

```diff
@@ -0,0 +1,56 @@
+package dns
+
+import (
+	"fmt"
+	"github.com/openshift/cluster-ingress-operator/pkg/dns"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	iov1 "github.com/openshift/cluster-ingress-operator/pkg/api/v1"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"testing"
+	"time"
+)
+
+func TestPublishRecordToZones(t *testing.T) {
+
+	var tests = []struct {
+		name   string
+		record *iov1.DNSRecord
+		zone   []configv1.DNSZone
+		result reconcile.Result
+		expect (operatorv1.ConditionStatus)
+	}{
+		{
+			name: "check status",
+			record: &iov1.DNSRecord{
+				Spec: iov1.DNSRecordSpec{
+					DNSName:    "subdomain.dnszone.io.",
+					RecordType: iov1.ARecordType,
+					Targets:    []string{"55.11.22.33"},
+				},
+			},
+			zone: []configv1.DNSZone{{
+				ID: "/subscriptions/E540B02D-5CCE-4D47-A13B-EB05A19D696E/resourceGroups/test-rg/providers/Microsoft.Network/dnszones/dnszone.io"},
+			},
+			result: reconcile.Result{RequeueAfter: 10 * time.Second},
+			expect: operatorv1.ConditionFalse,
+		},
+	}
+	for _, test := range tests {
+		r := &reconciler{
+			dnsProvider: &dns.FakeProvider{},
+		}
+		t.Logf("evaluating test %s", test.name)
+		actual, results := r.publishRecordToZones(test.zone, test.record)
+		log.Info("Actual")
+		log.Info(fmt.Sprintf("%#v", actual[0].Conditions[0].Status))
+		log.Info("Result")
+		log.Info(fmt.Sprintf("%#v", results))
```

Comment:
No need for all the logging.  Delete the `t.Logf` and `log.Info` lines, and add `test.name` to the `t.Fatalf` below (prefix `%q: ` to the format string).